collect/src/main/kotlin/com/certora/collect/EmptyTreapMap.kt

```diff
@@ -19,10 +19,19 @@ internal class EmptyTreapMap<@Treapable K, V> private constructor() : TreapMap<K
     override fun remove(key: K): TreapMap<K, V> = this
     override fun remove(key: K, value: V): TreapMap<K, V> = this
 
-    override fun updateValues(transform: (K, V) -> V?): TreapMap<K, V> = this
-    override fun parallelUpdateValues(parallelThresholdLog2: Int, transform: (K, V) -> V?): TreapMap<K, V> = this
+    override fun <R : Any> updateValues(
+        transform: (K, V) -> R?
+    ): TreapMap<K, R> = treapMapOf()
 
-    override fun <U> updateEntry(key: K, value: U?, merger: (V?, U?) -> V?): TreapMap<K, V> = 
+    override fun <R : Any> parallelUpdateValues(
+        parallelThresholdLog2: Int,
+        transform: (K, V) -> R?
+    ): TreapMap<K, R> = treapMapOf()
```

Comment:
can't you return `this` and just unsafe cast it? That's ultimately what `treapMapOf` is doing right? I'm not suggesting we do this, just making sure I understand what the code is doing.

Reply:
Yes, this is just a nicer-looking way of doing that.